ScarletEngine/include/Core/CoreUtils.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "Logger.h"
+
+// include signal.h to raise a debug break in POSIX systems (non-windows)
+#ifndef _MSC_VER
```

Comment:
only out of curiosity, what is this _MSC_VER macro? I wanna learn how you're setting things up to learn :)

Reply:
_MSC_VER is defined by the Microsoft MSVC compiler (it stores the current version). Basically this will check if it is _not_ being compiled by a Microsoft compiler and if so, include the POSIX "signal.h" header.